tests/components/onvif/test_config_flow.py

```diff
@@ -0,0 +1,508 @@
+                config_flow.CONF_HOST: HOST,
+                config_flow.CONF_PORT: PORT,
+                config_flow.CONF_USERNAME: USERNAME,
+                config_flow.CONF_PASSWORD: PASSWORD,
+            },
+        )
+
+        assert result["type"] == data_entry_flow.RESULT_TYPE_ABORT
+        assert result["reason"] == "no_h264"
+
+
+async def test_flow_import_onvif_api_error(hass):
+    """Test that config flow fails when ONVIF API fails."""
+    with patch("homeassistant.components.onvif.config_flow.get_device") as mock_device:
+        setup_mock_onvif_device(mock_device)
+        mock_device.create_devicemgmt_service = MagicMock(
+            side_effect=ONVIFError("Could not get device mgmt service")
+        )
+
+        result = await hass.config_entries.flow.async_init(
+            config_flow.DOMAIN,
+            context={"source": config_entries.SOURCE_IMPORT},
+            data={
+                config_flow.CONF_NAME: NAME,
+                config_flow.CONF_HOST: HOST,
+                config_flow.CONF_PORT: PORT,
+                config_flow.CONF_USERNAME: USERNAME,
+                config_flow.CONF_PASSWORD: PASSWORD,
+            },
+        )
+
+        assert result["type"] == data_entry_flow.RESULT_TYPE_ABORT
+        assert result["reason"] == "onvif_error"
+
+
+async def test_flow_import_onvif_auth_error(hass):
+    """Test that config flow fails when ONVIF API fails."""
+    with patch("homeassistant.components.onvif.config_flow.get_device") as mock_device:
+        setup_mock_onvif_device(mock_device)
+        mock_device.create_devicemgmt_service = MagicMock(
+            side_effect=Fault("Auth Error")
+        )
+
+        result = await hass.config_entries.flow.async_init(
+            config_flow.DOMAIN,
+            context={"source": config_entries.SOURCE_IMPORT},
+            data={
+                config_flow.CONF_NAME: NAME,
+                config_flow.CONF_HOST: HOST,
+                config_flow.CONF_PORT: PORT,
+                config_flow.CONF_USERNAME: USERNAME,
+                config_flow.CONF_PASSWORD: PASSWORD,
+            },
+        )
+
+        assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+        assert result["step_id"] == "auth"
+        assert result["errors"]["base"] == "connection_failed"
+
+
+async def test_option_flow(hass):
+    """Test config flow options."""
+    entry = await setup_onvif_integration(hass)
+
+    result = await hass.config_entries.options.async_init(entry.entry_id)
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["step_id"] == "onvif_devices"
+
+    result = await hass.config_entries.options.async_configure(
+        result["flow_id"],
+        user_input={
+            config_flow.CONF_EXTRA_ARGUMENTS: "",
+            config_flow.CONF_RTSP_TRANSPORT: config_flow.RTSP_TRANS_PROTOCOLS[1],
+        },
+    )
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5453535)